test/integration/offline.js

```diff
@@ -53,66 +53,56 @@ describe('offline', function () {
     complete.docs_written.should.equal(0)
   }).timeout(3 * 1000)
 
-  it('can\'t replication with live option.', async function () {
-    let err = null
-
-    try {
-      cozy.client.offline.createDatabase(DOCTYPE, {adapter: 'memory'})
-      await cozy.client.offline.replicateFromCozy(DOCTYPE, {live: true})
-    } catch (e) {
-      err = e
-    } finally {
-      err.should.eql(new Error('You can\'t use `live` option with Cozy couchdb.'))
-    }
+  it('can\'t replicate with live option.', async function () {
+    cozy.client.offline.createDatabase(DOCTYPE, {adapter: 'memory'})
+    return cozy.client.offline.replicateFromCozy(DOCTYPE, {live: true}).should.be.rejectedWith({ message: 'You can\'t use `live` option with Cozy couchdb.' })
   })
 
-  it('can auto replicate database', async function () {
-    let db = cozy.client.offline.createDatabase(DOCTYPE, {adapter: 'memory'})
-    const lastSeq = await new Promise((resolve) => {
-      cozy.client.offline.replicateFromCozy(DOCTYPE, {}, {complete: (info) => {
-        db.changes().on('complete', (info) => {
-          resolve(info.last_seq)
-        })
-      }})
-    })
-
-    let count = 0
-    db.changes({live: true, include_docs: true, since: lastSeq})
-      .on('change', (change) => { count++ })
-
-    const smallDoc = {test: 187}
-    cozy.client.offline.startSync(DOCTYPE, 3)
-    await new Promise(resolve => setTimeout(resolve, 1 * 1000))
-    let doc = await cozy.client.data.create(DOCTYPE, smallDoc)
-    const docId = doc._id
-
-    let err = null
-    doc = null
-
-    try {
-      doc = await db.get(docId)
-    } catch (e) {
-      err = e
-      err.status.should.be.exactly(404)
-    } finally {
-      (doc === null).should.be.true
-    }
+  it('can\'t replicate without adding the correct doctype.', async function () {
+    cozy.client.offline.createDatabase(DOCTYPE, {adapter: 'memory'})
+    const wrongDoctype = 'another.doctype'
+    return cozy.client.offline.replicateFromCozy(wrongDoctype).should.be.rejectedWith({ message: `You should add this doctype: ${wrongDoctype} to offline.` })
+  })
 
-    await new Promise(resolve => setTimeout(resolve, 5 * 1000))
+  it('can\'t replicate without creating the database in offline.', async function () {
+    const someDoctype = 'some.doctype'
+    return cozy.client.offline.replicateFromCozy(someDoctype).should.be.rejectedWith({ message: `You should add this doctype: ${someDoctype} to offline.` })
+  })
 
-    doc = null
-    err = null
+  it('can replicate created object in local database', async function () {
+    // create a database
+    const db = cozy.client.offline.createDatabase(DOCTYPE, { adapter: 'memory' })
+    // create a doc
+    const sampleDoc = { data: 'some Data' }
+    const remoteDoc = await cozy.client.data.create(DOCTYPE, sampleDoc)
```

Comment:
We need to delete the doc at the end of the test (in the `after()` function). So we could add `remoteDoc` to the global `docs` array.